multiaddr_test.go

```diff
@@ -923,3 +924,36 @@ func TestDNS(t *testing.T) {
 		t.Fatal("expected equality")
 	}
 }
+
+func TestHTTPPath(t *testing.T) {
+	t.Run("bad addr", func(t *testing.T) {
+		badAddr := "/http-path/thisIsMissingAfullBytes%f"
+		_, err := NewMultiaddr(badAddr)
```

Comment:
Can we add a test case for 
`/http-path/X/Y` If I understand it correctly for the returned multiaddr `addr.ValueForProtocol(ma.P_HTTP_PATH)` should have the value `X`